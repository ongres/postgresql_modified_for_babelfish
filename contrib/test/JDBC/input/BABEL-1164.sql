-- Test inserting less values than columns
CREATE TABLE t1(c1 int, c2 numeric, c3 varchar(32));
GO

INSERT INTO t1 VALUES (1, 2.0, 'hello');
GO

-- Expect error
INSERT INTO t1 VALUES (1, 2.0);
GO

SELECT * FROM t1;
GO

CREATE TABLE t2(id int IDENTITY, c1 int, c2 numeric, c3 AS c1 * c2);
GO

INSERT INTO t2 VALUES (1, 2.0);
GO

-- Expect error
INSERT INTO t2 VALUES (5);
GO

SELECT * FROM t2;
GO

INSERT INTO t1 SELECT c1, c2, c3 FROM t2;
GO

SELECT * FROM t1;
GO

-- Expect error
INSERT INTO t1 SELECT id, c2 FROM t2;
GO

SELECT * FROM t1;
GO

CREATE TABLE t3(c1 int, c2 numeric);
GO

INSERT INTO t3 VALUES (2, 4);
GO

-- Expect error
INSERT INTO t1 SELECT * FROM t3;
GO

DROP TABLE t1, t2, t3;
GO
