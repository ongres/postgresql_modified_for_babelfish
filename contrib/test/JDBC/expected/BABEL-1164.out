-- Test inserting less values than columns
CREATE TABLE t1(c1 int, c2 numeric, c3 varchar(32));
GO

INSERT INTO t1 VALUES (1, 2.0, 'hello');
GO
~~ROW COUNT: 1~~


-- Expect error
INSERT INTO t1 VALUES (1, 2.0);
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Number of given values does not match target table definition)~~


SELECT * FROM t1;
GO
~~START~~
int#!#numeric#!#varchar
1#!#2#!#hello
~~END~~


CREATE TABLE t2(id int IDENTITY, c1 int, c2 numeric, c3 AS c1 * c2);
GO

INSERT INTO t2 VALUES (1, 2.0);
GO
~~ROW COUNT: 1~~


-- Expect error
INSERT INTO t2 VALUES (5);
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Number of given values does not match target table definition)~~


SELECT * FROM t2;
GO
~~START~~
int#!#int#!#numeric#!#numeric
1#!#1#!#2#!#2.00000000
~~END~~


INSERT INTO t1 SELECT c1, c2, c3 FROM t2;
GO
~~ROW COUNT: 1~~


SELECT * FROM t1;
GO
~~START~~
int#!#numeric#!#varchar
1#!#2#!#hello
1#!#2#!#2
~~END~~


-- Expect error
INSERT INTO t1 SELECT id, c2 FROM t2;
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Number of given values does not match target table definition)~~


SELECT * FROM t1;
GO
~~START~~
int#!#numeric#!#varchar
1#!#2#!#hello
1#!#2#!#2
~~END~~


CREATE TABLE t3(c1 int, c2 numeric);
GO

INSERT INTO t3 VALUES (2, 4);
GO
~~ROW COUNT: 1~~


-- Expect error
INSERT INTO t1 SELECT * FROM t3;
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Number of given values does not match target table definition)~~


DROP TABLE t1, t2, t3;
GO
